soap/soap_utils.bal

```diff
@@ -179,8 +194,8 @@ function createSoapBody(xml payload, SoapVersion soapVersion) returns xml {
 # + return - The SOAP Request as `http:Request` with the SOAP envelope
 function fillSOAPEnvelope(SoapRequest request, SoapVersion soapVersion) returns http:Request {
     xml soapPayload = createSoapHeader(request, soapVersion);
-    xml requestPayload = request.payload;
-    if (!requestPayload.isEmpty()) {
+    var requestPayload = request["payload"];
+    if (requestPayload is xml) {
         xml body = createSoapBody(requestPayload, soapVersion);
         soapPayload += body;
     }
```

Comment:
Shall we handle the error also if the payload is not xml?

Reply:
This is not an error scenario. We create the xml soap payload if and only if the user passes the payload. Otherwise we don't append the `<soap:Body xmlns:soap="{{namespace}}"></soap:Body>`
part to xml payload which we are populating internally, in order to make the actual request.